### PR TITLE
improve e2e tests

### PR DESCRIPTION
**Pull request by petrjasek**
creating a branch so that I can see it on bamboo..

### CODE REVIEW

**Comment by petrjasek** [comment]
having some issues with `.travis.yml`, any idea what's wrong there @actionless ?

File: .travis.yml [hunk 1]
```diff
@@ -47,10 +47,13 @@ install:
...
-        env SUPERDESK_URL='http://127.0.0.1:5000/api' SUPERDESK_CLIENT_URL='http://127.0.0.1:9000' SUPERDESK_TESTING='True' honcho start &
-        cd client &&
-        grunt server:travis --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100' &
+        cd server
+        && (env SUPERDESK_URL='http://127.0.0.1:5000/api' SUPERDESK_CLIENT_URL='http://127.0.0.1:9000' SUPERDESK_TESTING='True' honcho start &)
+        && cd ../client
+        && grunt build --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100'
+        && cd dist
+        && while [ ! -f index.html ]; do sleep .5; done
+        && (python3 -m http.server 9000 &)
```
Comment (actionless) [inline .travis.yml]: also u need to go back (ie `cd ..`) after, it's all executed as one script, so next command will be executed inside `client/dist`

File: .travis.yml [hunk 2]
```diff
@@ -42,15 +42,19 @@ install:
...
+        cd server
         env SUPERDESK_URL='http://127.0.0.1:5000/api' SUPERDESK_CLIENT_URL='http://127.0.0.1:9000' SUPERDESK_TESTING='True' honcho start &
-        cd client &&
-        grunt server:travis --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100' &
+        cd ../client
+        grunt build --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100'
+        cd dist
+        while [ ! -f index.html ]; do sleep .5; done
+        python3 -m http.server 9000 &
+        cd ../..
```
Comment (actionless) [inline .travis.yml]: ;

File: .travis.yml [hunk 3]
```diff
@@ -42,15 +42,19 @@ install:
...
     - if [ "${TARGET}" != "${TARGET/e2e/}" ]; then
-        cd server &&
+        cd server
         env SUPERDESK_URL='http://127.0.0.1:5000/api' SUPERDESK_CLIENT_URL='http://127.0.0.1:9000' SUPERDESK_TESTING='True' honcho start &
-        cd client &&
-        grunt server:travis --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100' &
+        cd ../client
+        grunt build --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100'
+        cd dist
+        while [ ! -f index.html ]; do sleep .5; done
```
Comment (actionless) [inline .travis.yml]: ;

File: .travis.yml [hunk 4]
```diff
@@ -42,15 +42,19 @@ install:
...
 before_script:
     - if [ "${TARGET}" != "${TARGET/e2e/}" ]; then
-        cd server &&
+        cd server
         env SUPERDESK_URL='http://127.0.0.1:5000/api' SUPERDESK_CLIENT_URL='http://127.0.0.1:9000' SUPERDESK_TESTING='True' honcho start &
-        cd client &&
-        grunt server:travis --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100' &
+        cd ../client
+        grunt build --server='http://127.0.0.1:5000/api' --ws='ws://127.0.0.1:5100'
+        cd dist
```
Comment (actionless) [inline .travis.yml]: ; or &&

**Comment by actionless** [comment]
ah, actually each command should be separated by `&&` or `;` in depend of what needed.

it's all treated as one line in travis

**Comment by mugurrus** [comment]
:+1:

**Comment by petrjasek** [comment]
it seems to work quite good on bamboo atm - 70% green builds.
should I squash the commits into 1 or does it make sense to keep it separate?

**Comment by actionless** [comment]
imho better to squash

**Comment by petrjasek** [comment]
pls review @superdesk/developers

**Comment by amagdas** [comment]
:+1:

**Pull request merged**